src/runtime/asm_arm64.s

```diff
@@ -15,17 +15,28 @@ TEXT runtime·rt0_go(SB),NOSPLIT,$0
 	MOVW	R0, 8(RSP) // argc
 	MOVD	R1, 16(RSP) // argv
 
-	// create istack out of the given (operating system) stack.
-	// _cgo_init may update stackguard.
+	
 	MOVD	$runtime·g0(SB), g
+	// set the per-goroutine and per-mach "registers"
```

Comment:
Just curious, why was this code moved here from the nocgo label?

Reply:
The runtime·alloc_tls function (for windows) needs this before being called. My hunch is that moving this here should not affect other architectures but I could very easily be wrong here. 

The alternative is to make a separate pipeline for windows using conditionals and preserve the order of execution for other archs (as is) but it comes at the cost of code repetition. For the purpose of simplicity, I might just do this so I can avoid having to test other archs for now. We can perhaps revisit this later (when upstreaming)? Let me know your thoughts (changed this for now).